code/datums/components/laser_components.dm

```diff
@@ -0,0 +1,47 @@
+/datum/component/extralasers
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/lens_path
+	var/obj/item/ammo_casing/energy/ammo
+	var/user
+
+/datum/component/extralasers/Initialize(_ammo, _lens_path, lens)
+	if(!istype(parent, /obj/item/gun/energy/laser))
+		return COMPONENT_INCOMPATIBLE
+	attach(_ammo, _lens_path, lens)
+
+/datum/component/extralasers/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attackby)
+
+/datum/component/extralasers/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
+
+/datum/component/extralasers/proc/attackby(datum/source, obj/item/I, mob/user, params)
+	. = ..()
+	if(I.tool_behaviour == TOOL_CROWBAR)
+		detach(source, user, TRUE)
+
+/datum/component/extralasers/proc/attach(obj/item/ammo_casing/energy/_ammo, _lens_path, lens)
+	var/obj/item/gun/energy/laser/L = parent
+	ammo =  new _ammo (src)
+	L.ammo_type  += ammo
```

Comment:
Isn't this supposed to be types not instances?

Reply:
ammo_type is made of instances

Reply:
I just looked at it's types

Reply:
https://github.com/tgstation/tgstation/blob/a8fa220f7fd601f206363f68fff580b731edc0d4/code/modules/projectiles/guns/energy.dm#L53

Reply:
Ugh, why are only energy weapons storing instances